runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -502,9 +502,10 @@ ClassFileOracle::walkAttributes()
 			break;
 		}
 		case CFR_ATTRIBUTE_PermittedSubclasses: {
-			/* PermittedSubclasses verification is for Java 15 preview only. Don't record the attribute for other class versions
-			 * since it may be corrupt. */
-			if ((59 == _classFile->majorVersion) && (0 < _classFile->minorVersion)) {
+			/* PermittedSubclasses verification is for Java version >= 15 */
+			if ((classfile->majorVersion > 59)
+			|| ((59 == classfile->majorVersion) && (0 < classfile->minorVersion))
```

Comment:
Please dont use `0 <` - it just makes things harder to read.

Reply:
Isn't this check a very long winded way of saying `>= 59` ? Minor version is unsigned, so it's always going to be >= 0.

Reply:
For a preview feature, the minor version has to have all bits set:
> A class file denotes that it depends on the preview features of Java SE $N by having a major_version item that corresponds to Java SE $N and a minor_version item that has all 16 bits set. For example, a class file that depends on the preview features of Java SE 14 would have version 58.65535.

Not sure why we used `0 <` to start with.

Reply:
My comment about the code stands - it's not doing anything more than >= 59 would accomplish.

Reply:
Should the check be major == 59 and minor == 65535?

Reply:
That's a better check.  The attribute isn't valid in JDK15 by default - only in these specially marked classes

Reply:
Updated as suggested by GAC.